File: scripts/query_locally.R
library(DBI)
library(duckdb)
library(tictoc)

con <- dbConnect(duckdb::duckdb())

tic()
df <- dbGetQuery(con, "
    select interpreted.* from read_parquet('../occurrence/*.parquet')
    where interpreted.genus == 'Abra'
")
print(dim(df))
toc()

tic()
df <- dbGetQuery(con, "
    install spatial;
    load spatial;
    select interpreted.* from read_parquet('../occurrence/*.parquet')
    where ST_Intersects(geometry, ST_GeomFromText('POLYGON ((2.831383 51.212045, 2.896957 51.212045, 2.896957 51.240211, 2.831383 51.240211, 2.831383 51.212045))'))
")
print(dim(df))
toc()

dbDisconnect(con, shutdown = TRUE)

File: scripts/query_locally.py
import duckdb
import time

start_time = time.time()

query = """
    install spatial;
    load spatial;
    select
        dataset_id, _id as id,
        interpreted.*,
        source.scientificName as originalScientificName,
        flags,
        dropped,
        absence
    from read_parquet('s3://obis-open-data/occurrence/*.parquet')
    -- from read_parquet('/Users/pieter/Desktop/occurrence/*.parquet')
    where (interpreted.speciesid = 141433) 
        and (dropped is not true) 
        and (absence is not true);
"""

results = duckdb.query(query).to_df()
print(results)

end_time = time.time()
elapsed_time = end_time - start_time
print(f"Script executed in {elapsed_time:.2f} seconds")
